training-react/src/redux/user/reducer.js

```diff
@@ -1,18 +1,31 @@
+import { completeReducer, createReducer } from 'redux-recompose';
+
 import { actions } from './actions';
 
-const initialState = {};
+const USER_SESSION = 'USER_SESSION';
+
+const initialState = {
+  user: null
+};
 
-function reducer(state = initialState, action) {
-  switch (action.type) {
-    case actions.GET_USER:
-      return { ...state };
-    case actions.GET_USER_SUCCESS:
-      return { ...state, user: action.payload.user };
-    case actions.GET_USER_FAILURE:
-      return { ...state, error: action.payload };
-    default:
-      return state;
+const reducerDescription = {
+  primaryActions: [actions.GET_USER, actions.SET_USER],
+  override: {
+    [actions.GET_USER_SUCCESS]: (state, action) => {
+      const user = action.payload[0];
+      localStorage.setItem(USER_SESSION, JSON.stringify(user));
+      return { ...state, user };
+    },
+    [actions.GET_USER_FAILURE]: (state, action) => {
```

Comment:
Should we override `GET_USER_FAILURE` too?
If i remember correctly, we did this because `user` was inside `action.payload[0]`, but for the `error` it should work ok.
Let me know if i'm wrong